crm_rma_prodlot_supplier/models/stock_transfer_details.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright 2015 Vauxoo
+#    Author: Yanina Aular, Osval Reyes
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import api, models
+
+
+class StockTransferDetails(models.TransientModel):
+
+    _inherit = 'stock.transfer_details'
+
+    @api.multi
+    def do_detailed_transfer(self):
+        """
+        When incoming type transfer are made and stock move have serial/lot
+        number, the supplier is assigned to the serial/lot number taken from
+        picking.
+        @return: do_detailed_transfer boolean results
+        """
+        stock_prod = self.env['stock.production.lot']
+        stock_picking = self.env['stock.picking']
+        picking_brw = stock_picking.browse(self.picking_id.id)
+        if picking_brw.picking_type_id.code == 'incoming':
+            for lstits in [self.item_ids, self.packop_ids]:
+                for prod in lstits:
+                    if prod.lot_id:
+                        stock_prod_brw = stock_prod.browse(prod.lot_id.id)
+                        if not stock_prod_brw.supplier_id:
+                            stock_prod_brw.write({'supplier_id':
+                                                  picking_brw.partner_id.id})
+
+                        for move_line in picking_brw.move_lines:
```

Comment:
@pedrobaeza I will try to explain as best as possible.

1. I need to know the supplier invoice line for to calculate warranty with the supplier in case of a claim with it.
2. Then, supplier_invoice_line is created in stock.production.lot model of odoo.
3. we must find a way to obtain this supplier invoice line, we did not see another best way that assign it when it make a transfer of incoming type. if you know a way best that this, you explain me please.
4. Now, we used picking_brw.move_lines because stock move have link to the purchase order line, and, we find the invoice line across of purchase order line, but, i need to know that the product of specific serial number is the same that the product of move. I tell you the same in default_get, It's possible that this code can be improved.... @osvalr do you help me? please. When you're improving this code, let me know.

The improvements can be anything like @gurneyalex's comments: https://github.com/OCA/rma/pull/43#discussion-diff-40545297 https://github.com/OCA/rma/pull/43#discussion-diff-40545221 https://github.com/OCA/rma/pull/43#discussion-diff-40544890 https://github.com/OCA/rma/pull/43#discussion-diff-40546053

Thanks @gurneyalex for review the code

FYI @nhomar.